src/pivot-table/hooks/use-column-width.ts

```diff
@@ -102,41 +102,62 @@ export default function useColumnWidth(
             return getPercentageValue(columnWidth.percentage) * rect.width;
+          case ColumnWidthType.FitToContent:
+          case ColumnWidthType.Auto:
           default:
-            // fit to content / auto
             return fitToContentWidth;
         }
       };
 
+      const maxMeasureCellWidth = qMeasureInfo.reduce((maxWidth, { qFallbackTitle, columnWidth }) => {
+        const fitToContentWidth = measureTextForMeasureValue(qFallbackTitle) + TOTAL_CELL_PADDING;
+        return Math.max(maxWidth, getColumnWidth(columnWidth, fitToContentWidth));
+      }, 0);
+
       let sumOfWidths = 0;
 
-      const lastRow = headersData.data[headersData.size.y - 1] as HeaderCell[];
-      const columnWidths = lastRow.map((cell, index) => {
-        let width: number;
+      const lastRow = headersData.data.at(-1) as HeaderCell[];
+      const columnWidths = lastRow.map((_, collIdx) => {
+        let width = TOTAL_CELL_PADDING;
 
-        if (widthOverride && overrideIndex !== undefined && overrideIndex === index) {
+        if (widthOverride && overrideIndex !== undefined && overrideIndex === collIdx) {
           width = widthOverride;
-        } else if (cell.id === PSEUDO_DIMENSION_KEY) {
-          // Use the max width of all measures
-          width = Math.max(
-            ...qMeasureInfo.map(({ qFallbackTitle, columnWidth }) => {
-              const fitToContentWidth = measureTextForDimensionValue(qFallbackTitle) + TOTAL_CELL_PADDING;
-              return getColumnWidth(columnWidth, fitToContentWidth);
-            }),
-          );
         } else {
-          const { label, qApprMaxGlyphCount, columnWidth, isLocked } = cell;
-          const expandIconSize = !isFullyExpanded && index < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
-          const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
-
-          const fitToContentWidth =
-            TOTAL_CELL_PADDING +
-            Math.max(
-              measureTextForHeader(label) + MENU_ICON_SIZE + lockedIconSize,
-              estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + expandIconSize,
-            );
-
-          width = getColumnWidth(columnWidth, fitToContentWidth);
+          width = headersData.data.reduce((maxWidth, row, rowIdx) => {
+            const header = row[collIdx];
+            if (!header) return maxWidth;
+            let cellWidth = 0;
+
+            if (header.id === PSEUDO_DIMENSION_KEY) {
+              // Use the max width of all measures
+              cellWidth = maxMeasureCellWidth;
+            } else {
+              const { label, qApprMaxGlyphCount, columnWidth, isLocked } = header;
+              const expandIconSize = !isFullyExpanded && collIdx < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
+              const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
+
+              let fitToContentWidth = TOTAL_CELL_PADDING;
+              if (layoutService.isLeftDimension(header.dimensionInfoIndex)) {
```

Comment:
Should be able to resolve and assign `layoutService.isLeftDimension(header.dimensionInfoIndex)` on the actual `header` in `extract-headers`. I will probably reduce the number of times it has to be resolved.

Reply:
Done